src/platform_impl/linux/wayland/seat/text_input/handlers.rs

```diff
@@ -31,6 +31,7 @@ pub(super) fn handle_text_input(
 
             // Enable text input on that surface.
             if window_handle.ime_allowed.get() {
+                text_input.set_content_type_by_purpose(window_handle.ime_purpose.get());
                 text_input.enable();
```

Comment:
From the protocol:

 `The set_surrounding_text, set_content_type and set_cursor_rectangle requests must follow if the text input supports the necessary functionality.`
 
Could you swap them, just to prevent potential weird behavior? While the updates to protocol are atomic on commit, I'd do that just to ensure.